### ISSUE TITLE
[suggestion] be able to preselect the first exact match

### ISSUE THREAD

**Issue by FractalWire** [junegunn/fzf#1703]
- Category
    - [X] scripting
- OS
    - [X] all
- Shell
    - [X] all

In a scripting perspective, I'm trying to preselect an entry on fzf when it starts.
That selection might be different from the `--query` option.
This option would allow fzf to keep context between two calls and could be called `--selection` for example.

Here a typical use case:

```bash
while true
do
    {read query; read key; read ret } <<< $(
        fzf --selection="$ret" \
            --print-query --query="$query" \
            --expect ctrl-h --preview "$cmd {}"
    )
    if test "$key" = 'ctrl-h'
    then
        cmd='somenewcommand'
    fi
done
```

Currently, the only way to pre-select an entry is to modify the query with the exact match. By doing that we lose the context of fzf, as both the query and the selection might matter.

**Comment by junegunn** [comment]
See my previous response to the suggestion: https://github.com/junegunn/fzf/issues/1296#issuecomment-394560027. And also #395.

The point is, making fzf a fully-featured building block for terminal applications was never a goal to me. Even though there are ambitious users who try to implement all kinds of things with fzf, I'd like to keep fzf conceptually simple and small; an interactive Unix filter.

It's not clear to me what you're trying to achieve here (dynamically changing the preview command?), but I'd consider rearranging the order of input lines before feeding it to fzf, i.e. feed "preselected" entry as the first line. Or you could provide `execute` bindings to give extra options to "preview" the entry differently.

```sh
fzf --bind 'ctrl-h:execute:less {} < /dev/tty > /dev/tty'
```

**Comment by FractalWire** [comment]
Yes, I get your idea of simplicity over feature-complete for fzf. This is one of the reason this tool is so powerful.
I'll try to petition this feature a bit more none the less. Especially given that I actually implemented it yesterday in fzf.

> It's not clear to me what you're trying to achieve here (dynamically changing the preview command?)

I'm trying exactly that. To give a bit more context, it's about building a fuzzy file explorer. We are discussing it here : https://github.com/junegunn/fzf/issues/1679

In fact, in my use case, I need to pass a variable to the preview command for it to change. Given that the preview command looks like a black box, the easiest way to achieve that is to restart fzf with a preview command changed slightly.
It works fine, but I loose the cursor position, which, in my use case, is annoying.

>  I'd consider rearranging the order of input lines before feeding it to fzf, i.e. feed "preselected" entry as the first line

I tried that, but it is not satisfactory for me. keeping the cursor position is important, as you have similar enough entry around the one selected.

Now, the way I implemented it is a bit different from the way I asked for it. I don't select the first match of a string, but I select an entry at a particular index.
That index might be given by a similar option as `--print-query`, named `--print-sel-idx`, or it might be obtained by parsing the input and getting the line you're interested in, before feeding it to fzf.
I'll make a PR, just so that we can discuss this a bit further if you'd like: if it's simple enough for you or if you see some corner cases not covered by my implementation.

This change is just a few lines long, so I don't think it introduces a lot of new code complexity.
This is definitely an advanced scripting feature but it should not affect the way fzf works otherwise.
